pkg/asset/ignition/bootstrap/baremetal/ironic_creds.go

```diff
@@ -0,0 +1,56 @@
+package baremetal
+
+import (
+	"crypto/rand"
+	"math/big"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+// IronicCreds is the asset for the ironic user credentials
+type IronicCreds struct {
+	Username string
+	Password string
+}
+
+var _ asset.Asset = (*IronicCreds)(nil)
+
+// Dependencies returns no dependencies.
+func (a *IronicCreds) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate the kubeadmin password
+func (a *IronicCreds) Generate(asset.Parents) error {
+	pw, err := generateRandomPassword()
+	if err != nil {
+		return err
+	}
+	a.Username = "bootstrap-user"
+	if a.Password == "" {
```

Comment:
I understand that this was copied from the kubeadmin password asset, but there should not be a check for whether `a.Password` is empty. The `Generate` function should always generate everything fresh.